星穹铁道的便笺

后续有计划做星穹铁道的便笺吗？我尝试自己去添加，结果返回的内容是
```
07-17 15:38:25 [DEBUG] nonebot_plugin_mystool | 网络请求返回: {"data":null,"message":"invalid request","retcode":-10001}
``` 
我添加了
```
URL_STARRAIL_STATUS_BBS = "https://api-takumi-record.mihoyo.com/game_record/app/hkrpg/api/note"
URL_STARRAIL_STATUS_WIDGET = "https://api-takumi-record.mihoyo.com/game_record/app/hkrpg/aapi/widget"

async def StarRail_board(account: UserAccount) -> Tuple[
    Union[BaseApiStatus, StarRailBoardStatus], Optional[StarRailBoard]]:
    """
    获取星铁实时便笺

    :param account: 用户账户数据
    """
    game_record_status, records = await get_game_record(account)
    if not game_record_status:
        return game_record_status, None
    game_list_status, game_list = await get_game_list()
    if not game_list_status:
        return game_list_status, None
    game_filter = filter(lambda x: x.en_name == 'sr', game_list)
    game_info = next(game_filter, None)
    if not game_info:
        return StarRailBoardStatus(no_starrail_account=True), None
    else:
        game_id = game_info.id
    flag = True
    for record in records:
        if record.game_id == game_id:
            try:
                flag = False
                params = {"role_id": record.game_role_id, "server": record.region}
                print("params:",params)
                url = f"{URL_STARRAIL_STATUS_BBS}?{urlencode(params)}"
                print(url)
                headers = HEADERS_STARRAIL_STATUS_BBS.copy()
                headers["x-rpc-device_id"] = account.device_id_android
                async for attempt in get_async_retry(False):
                    with attempt:
                        headers["DS"] = generate_ds(
                            params={"role_id": record.game_role_id, "server": record.region})
                        print(headers)
                        async with httpx.AsyncClient() as client:
                            res = await client.get(url, headers=headers,
                                                   cookies=account.cookies.dict(v2_stoken=True, cookie_type=True),
                                                   timeout=_conf.preference.timeout)
                        api_result = ApiResultHandler(res.json())
                        print("第一次：",api_result)
                        if api_result.login_expired:
                            logger.info(
                                f"星铁实时便笺: 用户 {account.bbs_uid} 登录失效")
                            logger.debug(f"网络请求返回: {res.text}")
                            return StarRailBoardStatus(login_expired=True), None

                        if api_result.invalid_ds:
                            logger.info(
                                f"星铁实时便笺: 用户 {account.bbs_uid} DS 校验失败")
                            logger.debug(f"网络请求返回: {res.text}")
                        if api_result.retcode == 1034:
                            logger.info(
                                f"星铁实时便笺: 用户 {account.bbs_uid} 可能被验证码阻拦")
                            logger.debug(f"网络请求返回: {res.text}")
                        if api_result.invalid_ds or api_result.retcode == 1034:
                            headers["DS"] = generate_ds()
                            headers["x-rpc-device_id"] = account.device_id_ios
                            async with httpx.AsyncClient() as client:
                                res = await client.get(
                                    URL_STARRAIL_STATUS_WIDGET,
                                    headers=headers,
                                    cookies=account.cookies.dict(v2_stoken=True, cookie_type=True),
                                    timeout=_conf.preference.timeout
                                )
                            print("请求返回：",res)
                            api_result = ApiResultHandler(res.json())
                            print("第二次：",api_result)
                            return StarRailBoardStatus(success=True), \
                                StarRailBoard.parse_obj(api_result.data)
                        return StarRailBoardStatus(success=True), StarRailBoard.parse_obj(api_result.data)
            except tenacity.RetryError as e:
                if is_incorrect_return(e):
                    logger.exception(f"星铁实时便笺: 服务器没有正确返回")
                    logger.debug(f"网络请求返回: {res.text}")
                    return StarRailBoardStatus(incorrect_return=True), None
                else:
                    logger.exception(f"星铁实时便笺: 请求失败")
                    return StarRailBoardStatus(network_error=True), None
    if flag:
        return StarRailBoardStatus(no_starrail_account=True), None
``` 
game_list和game_records都成功返回内容，但是倒在了这里
```
async with httpx.AsyncClient() as client:
                            res = await client.get(url, headers=headers,
                                                   cookies=account.cookies.dict(v2_stoken=True, cookie_type=True),
                                                   timeout=_conf.preference.timeout)
                        api_result = ApiResultHandler(res.json())
``` 
下面是终端上的错误信息
```
07-17 15:38:20 [SUCCESS] nonebot | OneBot V11 3451066253 | [message.private.friend]: Message -892711309 from aaaaaaaa '/便笺sr'
07-17 15:38:20 [DEBUG] nonebot | Checking for matchers in priority 1...
07-17 15:38:20 [DEBUG] nonebot | Checking for matchers in priority 4...
07-17 15:38:20 [DEBUG] nonebot | Checking for matchers in priority 5...
07-17 15:38:20 [INFO] nonebot | Event will be handled by Matcher(type='message', module=nonebot_plugin_mystool.plan)
07-17 15:38:20 [DEBUG] nonebot | Running Matcher(type='message', module=nonebot_plugin_mystool.plan)
07-17 15:38:20 [TRACE] nonebot | Matcher(type='message', module=nonebot_plugin_mystool.plan) run with incoming args: bot=Bot(type='OneBot V11', self_id='3451066253'), event=PrivateMessageEvent(time=1689579500, self_id=3451066253, post_type='message', sub_type='friend', user_id=aaaaaaaa, message_type='private', message_id=-892711309, message=[MessageSegment(type='text', data={'text': '/便笺sr'})], original_message=[MessageSegment(type='text', data={'text': '/便笺sr'})], raw_message='/便笺sr', font=0, sender=Sender(user_id=aaaaaaaa, nickname='哥', sex='unknown', age=0, card=None, area=None, level=None, role=None, title=None), to_me=True, reply=None, target_id=3451066253), state={'_prefix': {'command': ('便笺sr',), 'raw_command': '/便笺sr', 'command_arg': [], 'command_start': '/', 'command_whitespace': None}}
07-17 15:38:20 [DEBUG] nonebot | Running handler Dependent(call=_)
---params: {'role_id': 'bbbbbbb', 'server': 'prod_gf_cn'}
---https://api-takumi-record.mihoyo.com/game_record/app/hkrpg/api/note?role_id=bbbbbbb&server=prod_gf_cn
---{'DS': '1689579501,193865,3b4a00e9fef2b03514d8e315646212d6', 'x-rpc-device_id': 'F9235676-FBF4-46E5-BB9E-A9330E0FDC88', 'Accept': 'application/json,text/plain,*/*', 'Origin': 'https://webstatic.mihoyo.com', 'User-agent': 'Mozilla/5.0 (Linux; Android 11; MI 8 SE Build/RQ3A.211001.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/104.0.5112.97 Mobile Safari/537.36 miHoYoBBS/2.36.1', 'Referer': 'https://webstatic.mihoyo.com/', 'x-rpc-app_version': '2.28.1', 'x-rpc-channel': 'ys', 'X-Requested-With': 'com.mihoyo.hyperion', 'x-rpc-client_type': '5'}
---第一次： content={'data': None, 'message': '', 'retcode': 1034} data=None message=None retcode=1034
07-17 15:38:22 [INFO] nonebot_plugin_mystool | 星铁实时便笺: 用户 76454149 可能被验证码阻拦
07-17 15:38:22 [DEBUG] nonebot_plugin_mystool | 网络请求返回: {"data":null,"message":"","retcode":1034}
---请求返回： <Response [200 OK]>
---第二次： content={'data': None, 'message': 'invalid request', 'retcode': -10001} data=None message='invalid request' retcode=-10001
07-17 15:38:24 [ERROR] nonebot_plugin_mystool | 星铁实时便笺: 服务器没有正确返回
Traceback (most recent call last):
  File "pydantic\main.py", line 522, in pydantic.main.BaseModel.parse_obj
    obj = dict(obj)
TypeError: 'NoneType' object is not iterable

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot_plugin_mystool\simple_api.py", line 1475, in StarRail_board
    StarRailBoard.parse_obj(api_result.data)
  File "pydantic\main.py", line 525, in pydantic.main.BaseModel.parse_obj
    raise ValidationError([ErrorWrapper(exc, loc=ROOT_KEY)], cls) from e
pydantic.error_wrappers.ValidationError: 1 validation error for StarRailBoard
__root__
  StarRailBoard expected dict not NoneType (type=type_error)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 17, in <module>
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot\__init__.py", line 309, in run
    get_driver().run(*args, **kwargs)
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot\drivers\fastapi.py", line 198, in run
    uvicorn.run(
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\uvicorn\main.py", line 578, in run
    server.run()
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\asyncio\base_events.py", line 640, in run_until_complete
    self.run_forever()
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\asyncio\base_events.py", line 607, in run_forever
    self._run_once()
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\asyncio\base_events.py", line 1922, in _run_once
    handle._run()
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot\message.py", line 467, in check_and_run_matcher
    await _run_matcher(
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot\message.py", line 419, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 753, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 728, in simple_run
    await handler(
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot\dependencies\__init__.py", line 108, in __call__
    return await cast(Callable[..., Awaitable[R]], self.call)(**values)
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot_plugin_mystool\plan.py", line 102, in _
    await resin_check_sr(bot=bot, qq=event.user_id, is_auto=False, group_event=event)
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot_plugin_mystool\plan.py", line 462, in resin_check_sr
    starrail_board_status, board = await StarRail_board(account)
> File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\nonebot_plugin_mystool\simple_api.py", line 1436, in StarRail_board
    async for attempt in get_async_retry(False):
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\tenacity\_asyncio.py", line 71, in __anext__
    do = self.iter(retry_state=self._retry_state)
  File "C:\ProgramData\anaconda3\envs\mhy\Lib\site-packages\tenacity\__init__.py", line 326, in iter
    raise retry_exc from fut.exception()
tenacity.RetryError: RetryError[<Future at 0x1f75acef690 state=finished raised ValidationError>]
07-17 15:38:25 [DEBUG] nonebot_plugin_mystool | 网络请求返回: {"data":null,"message":"invalid request","retcode":-10001}
07-17 15:38:25 [INFO] nonebot_plugin_mystool | success=False network_error=False incorrect_return=True login_expired=False need_verify=False invalid_ds=False no_starrail_account=False
07-17 15:38:25 [DEBUG] nonebot | OneBot V11 | Calling API send_private_msg
07-17 15:38:26 [INFO] nonebot | Matcher(type='message', module=nonebot_plugin_mystool.plan) running complete
07-17 15:38:26 [DEBUG] nonebot | Stop event propagation
07-17 15:38:26 [DEBUG] nonebot | Checking for matchers completed
```

--- a/src/nonebot_plugin_mystool/plan.py
+++ b/src/nonebot_plugin_mystool/plan.py
@@ -531,12 +531,21 @@ async def resin_check_sr(bot: Bot, qq: int, is_auto: bool,
                    f"\n⏳开拓力数量：{board.current_stamina} / 180" \
                    f"\n⏱开拓力将在{board.stamina_recover_text}回满" \
                    f"\n📒每日实训：{board.current_train_score} / {board.max_train_score}" \
-                   f"\n📅每日委托：{4 - board.accepted_expedition_num} 个任务未完成" \
+                   f"\n📅每日委托：{board.accepted_expedition_num} / 4" \
                    f"\n🌌模拟宇宙：{board.current_rogue_score} / {board.max_rogue_score}"
-            if group_event:
-                await bot.send(event=group_event, at_sender=True, message=msg)
+            if not is_auto:
+                if group_event:
+                    await bot.send(event=group_event, at_sender=True, message=msg)
+                else:
+                    await bot.send_private_msg(user_id=qq, message=msg)
             else:
-                await bot.send_private_msg(user_id=qq, message=msg)
+                if board.current_stamina >= _conf.preference.stamina_threshold:
+                    if group_event:
+                        await bot.send(event=group_event, at_sender=True, message=msg)
+                    else:
+                        await bot.send_private_msg(user_id=qq, message=msg)
+                else:
+                    logger.info(f"崩铁实时便笺：账户 {account.bbs_uid} 开拓力:{board.current_stamina},未满足推送条件")
 
 
 

--- a/src/nonebot_plugin_mystool/plugin_data.py
+++ b/src/nonebot_plugin_mystool/plugin_data.py
@@ -86,6 +86,8 @@ class Preference(BaseSettings):
     '''每日自动签到和米游社任务的定时任务执行时间，格式为HH:MM'''
     resin_interval: int = 60
     '''每次检查原神便笺间隔，单位为分钟'''
+    stamina_threshold: int = 0
+    '''崩铁便签开拓力提醒阈值，默认为0：一直提醒'''
     geetest_url: Optional[str]
     '''极验Geetest人机验证打码接口URL'''
     geetest_json: Optional[Dict[str, Any]] = {

